### PR TITLE
:sparkles:  Collection Activity Tab

### LINKED ISSUE #5272: Collection activity tab
Here we go again 

### Figma link and basics
- updated collection detail handoff figma: 
https://www.figma.com/file/7BWdGiowrvTucpSFLE0kRX/Collection-detail-handoff?node-id=0%3A1&t=J1caM6sKXhnhigJ7-1
- newly divided into two section, items and activity 
- activity frames have some sort of working prototypes
- new frame - 1024px wide representing tablet/small desktop view - better clarity for layout 

image

# Desktop
desktop

## Top section 
- remains the same 

## Menu bar 
- everything except the tab switch is removed when on activity 

## Burger menu 
- new are event types - used for sorting events 
- there is also filtering based traits etc for the future
- price I think won't be there, it's a leftover design

## Chart
- same as on gallery item
- gets smaller when resizing the screen, see 1024 screen 
- button for switching time frame - without drop shadow this time so it's not that disruptive

## Owner insights
- option to switch between - same hover effects and function as search tab switch
### Holders 
image
- this tab is new, with few changes to the hierarchy and layout to make it clearer
- the tab is open by default - you can then switch to flipper from there
- vertical scroll 
- the box is made for mobile by default so it imo should not get smaller
image
- name is like in gallery item detail - when you hover, change to hover color and hover card is displayed 
image

- stats, taken from current holder tab 
  - owned is amount of nfts 
  - total bought - in current version bought, but it did not make sense for this layout so its renamed 
  - total sold - same case as with total bought 
  - date - latest activity - no changes 

- nft details dropdown, gives you option to view the nfts of the holder, also taken from current tab but refined 
  - for holders we will only display the nft, name and date
  - hover over the section results in change of bg
 
	image
   - spacing can be seen in figma

### Flippers 
image

- layout similar to holders but filled with more information
- name same 
- stats same with small change that is the last activity and best flip - showed in percentage
- also with option to show nft details 
	
image
   
   - img
   - name
   - profit - percentage
   - bought for and sold for, if there is no data yet, two dashes will make the job 
- showed the hover effect over the section, see figma for exact padding etc 

## Event section
- we will have two types how to show event, one is for mobile and the other for desktop - they would be used based on the room available
- infinity scroll - no pagination, just scroll 'til infinity :D 

### Desktop
image

- first one is this - desktop 
- no hover effect over the row 
- you have the option to hover over: name, address from/to and time that will show exact time in the tooltip
- that will trigger hover card 
- breadcrumbs are on top of this all - showing active filters: event type or trait filter

### Collapsed view/mobile
image
image

- replacing the desktop view when there is no room available to maintain somewhat functional gaps between the info. 
- something like this is ready for switch: 

image

- imgs up represent the use cases - it can be also stretched if there is space

### Other event types
image

- this shows how are different events displayed - some are without adress, or with price, you name it

### Tags
- these are new
- many new colors used, they are not yet somewhere else except the k-bluelight
image
- in figma there is light and dark mode version - no hover effect since it's not a button 

# Mobile 
image

- holder and flipper tab is already mobile compatible so it just slides on top
- chart under it - smaller version, like in gallery item on mobile 
- then divider and events - collapsed version
- there are also going to be some changes coming to the mobile about the filter, because now you have to scroll all the way to the top to change event type
- rest is working the same

## Dark mode 
- no changes here, most of the components are well known - tags are in the figma and hover over container is also there as a reminder

As always, feel free to ask! 
Thank you 🚀

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
@yangwao  Deliverd on time?
👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5272 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [x] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

tested on these collections:

- [x] /rmrk/collection/40BC7893024C43370B-CAPYBARA/activity
- [x] /rmrk/collection/4A43156FE23E061C03-SNK/activity
- [x] /bsx/collection/945672150/activity
- [x] /bsx/collection/2551182625

### CODE REVIEW

**Comment by yangwao** [comment]
looks neat!

**Comment by daiagi** [comment]
@roiLeo @preschian  @Jarsen136 

i want to ask of you kindly:
1. please try to review this ASAP, I'm trying to win a race :laughing:  https://github.com/kodadot/nft-gallery/issues/5272#issuecomment-1479504428
2. again, because I'm trying to win the race, i ask you be kind with me in the reviews
 if you think the issue you spotted is a show stopper and no way it can be merged without fixing it first, so be it
 otherwise i ask that you will approve the PR and comment whatever it is that needs changing, I will make a follow up PR to address all these issues
 
 Thank you :heart:

File: components/collection/activity/Activity.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+useResizeObserver(wrapper, (entry) => {
+  if (entry[0].contentRect.width >= mobileBreakpoint) {
+    tablet.value = true
+    mobile.value = false
+  } else {
+    tablet.value = false
+    mobile.value = true
+  }
+})
```
Comment (Jarsen136) [inline components/collection/activity/Activity.vue:76]: pls only keep one state `isMobile` or `isTablet`

File: components/collection/activity/events/eventRow/common.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  return blank
+}
+
+export const getToAddress = (event: InteractionWithNFT | Offer): string => {
+  const interaction = event.interaction
+  if (interaction === Interaction.MINTNFT || interaction === Interaction.LIST) {
+    return event.caller
+  }
+  if (interaction === Interaction.BUY || interaction === Interaction.SEND) {
+    return event.caller
```
Comment (Jarsen136) [inline components/collection/activity/events/eventRow/common.ts:64]: should it return `caller` for the send event? I guess not

**Comment by exezbcz** [comment]
woho, lets go 🚀

File: components/collection/utils/timeAgo.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  const minutesSince = Math.floor(secondsSince / 60)
+  if (minutesSince < 60) {
+    return `${minutesSince} ${minutesSince !== 1 ? minutes : minute} ${ago}`
+  }
+  const hoursSince = Math.floor(minutesSince / 60)
+  if (hoursSince < 24) {
+    return `${hoursSince} ${hoursSince !== 1 ? hours : hour} ${ago}`
+  }
+  const daysSince = Math.floor(hoursSince / 24)
+  return `${daysSince} ${daysSince !== 1 ? days : day} ${ago}`
```
Comment (Jarsen136) [inline components/collection/utils/timeAgo.ts:28]: What about `months` and `years`? 
please use `formatDistanceToNow` utils function instead

`import { formatDistanceToNow } from 'date-fns'`

File: components/collection/utils/useCollectionActivity.ts [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+  return owners
+}
+
+const preProccessForFindingFlippers = (interactions: InteractionWithNFT[]) => {
+  const changeHandsInteractions: InteractionWithNFT[] = []
+  const NFTS: NFTMap = {}
+
+  interactions.forEach((event) => {
+    if (event.interaction === Interaction.MINTNFT) {
+      // mintInteractions.push(event)
```
Comment (Jarsen136) [inline components/collection/utils/useCollectionActivity.ts:101]:
```suggestion
```

File: components/collection/activity/events/eventRow/common.ts (hunk 2 above)
Comment (daiagi) [reply components/collection/activity/events/eventRow/common.ts:64]: correct. thank you. it should be meta

Comment (daiagi) [reply components/collection/activity/events/eventRow/common.ts:64]: Done. thank you

File: components/collection/activity/Activity.vue (hunk 1 above)
Comment (daiagi) [reply components/collection/activity/Activity.vue:76]: yep. thank you :)
fixed

File: components/collection/activity/events/eventRow/common.ts (hunk 2 above)
Comment (Jarsen136) [reply components/collection/activity/events/eventRow/common.ts:64]: Then we could put MINTNFT LIST BUY events in the same condition. I would prefer to use `switch case` statement

**Comment by exezbcz** [comment]
# Feedback
### name
image
- name if its black to bold - please pay attention to the blue links - so they dont change with this (so only the name should be bold, like in flipper tab) 

### divider
image
- to 40px please - on mobile 

### No hover card 
image
- there should be hover card, on the names in flipper/holder tab as well

### Tags
image
image

- that was wrongly formulated by me, but I would maybe lower the padding on the transfer tag - which is not that common - so we can then make the fixed width on other tags lower
- something like 8px works

### Alignment of the menu bar 
image
- it's not in line and when you switch from activity to items its even more noticable 

### Smol color change
image
- from/to to black please



# To follow up issue I guess
image
- we could maybe switch to the mobile layout of the event bit sooner 

scrolling position
- can we make it so the switch between items and activity does not reset the scroll position 
- it's better when its seamless

**Comment by exezbcz** [comment]
+ I would also remove the price filter from the burger menu

**Comment by yangwao** [comment]
> @yangwao Deliverd on time?

yes, it counts, let's see how good is it! :) 

dropping URL for easier visual testings 
https://deploy-preview-5345--koda-nuxt.netlify.app/rmrk/collection/40BC7893024C43370B-CAPYBARA/activity
https://deploy-preview-5345--koda-nuxt.netlify.app/rmrk/collection/4A43156FE23E061C03-SNK/activity
https://deploy-preview-5345--koda-nuxt.netlify.app/bsx/collection/945672150/activity
https://deploy-preview-5345--koda-nuxt.netlify.app/bsx/collection/2551182625/activity

**Comment by daiagi** [comment]
> * I would also remove the price filter from the burger menu

removed

**Comment by yangwao** [comment]
I see some dance at 1090px (two-thirds of 13")

wyt @exezbcz if we drop "KSM" from amount as it's currently repeating information ? Basilisk will have multicurrency support for NFTs (but iirc is not there yet possible)

image

for me is go time on desktop.

also I've realized now, @exezbcz collection offers is not there 👀 pushing to v2
- #5362

File: components/collection/utils/timeAgo.ts (hunk 3 above)
Comment (daiagi) [reply components/collection/utils/timeAgo.ts:28]: it doesn't fit the format in the figma file

let's handle it in a follow up issue ok?

File: components/collection/activity/events/eventRow/common.ts (hunk 2 above)
Comment (daiagi) [reply components/collection/activity/events/eventRow/common.ts:64]: noted.
with you permission let's handle in follow up

**Comment by daiagi** [comment]
> I see some dance at 1090px (two-thirds of 13")


That's weird, you are supposed to see the "mobile" format at that width

image

**Comment by yangwao** [comment]
> see the "mobile" format at that width

well, mobile width != mobile device, I already noticed detection is not so great sometimes :)


at https://deploy-preview-5345--koda-nuxt.netlify.app/rmrk2/collection/9ea8d4ad827797a01e-LETTER/activity

I guess chart lines should be not so connected twice in same direction from same datapoint? 
image

**Comment by daiagi** [comment]
> I guess chart lines should be not so connected twice in same direction from same datapoint?

Probably not indeed

**Comment by exezbcz** [comment]
@yangwao yeah possible, or add it next to the column tag - so there is amount(currency) 
still weird that in the screenshot above I have BSX as currency :D 
image


- I think it won't free up much space and if it gives more clarity to the user, I would leave it there. 

collection offer - 
image

**Comment by exezbcz** [comment]
the divider should have margin 40 only when breadcrumbs are disabled 
image

breadcrumbs should remain 24px from the divider 
events are 40px from the divider

**Comment by daiagi** [comment]
> collection offer - image

old bug, not to be solve din this PR. it is do do with us setting z-index:1000 on the entire page for the footer graphic to show well, that causes this issue with the tooltip

**Comment by exezbcz** [comment]
@daiagi yeah, dont solve it here - just made screenshot for wao that there is collection offer - but not implemented yet

**Comment by exezbcz** [comment]
there is still place in the container, is it possible to stretch that? 
image 

the date for example

- also would be good if we avoid names/ids jumping to the second row
- maybe add wrap text and then some dots? The full ID/name is visible in the hover card 
image

File: components/collection/utils/timeAgo.ts (hunk 3 above)
Comment (Jarsen136) [reply components/collection/utils/timeAgo.ts:28]: Oki, let's make a follow up issue. I guess we could define a custom time format with `date-fns`

File: components/collection/activity/events/eventRow/common.ts (hunk 2 above)
Comment (Jarsen136) [reply components/collection/activity/events/eventRow/common.ts:64]: It's a small change. At least we could write so please
```
if (interaction === Interaction.MINTNFT || interaction === Interaction.LIST || interaction === Interaction.BUY)  
{ 
xxx
}
```

Comment (daiagi) [reply components/collection/activity/events/eventRow/common.ts:64]: sure. ok.  will do

**Comment by daiagi** [comment]
> the divider should have margin 40 only when breadcrumbs are disabled image
> 
> breadcrumbs should remain 24px from the divider events are 40px from the divider

Fixed

File: components/collection/activity/events/eventRow/common.ts (hunk 2 above)
Comment (daiagi) [reply components/collection/activity/events/eventRow/common.ts:64]: done

**Comment by yangwao** [comment]
image

image


Also duplication and complexity notice  https://codeclimate.com/github/kodadot/nft-gallery/pull/5345 👀

**Comment by daiagi** [comment]
> image
> 
> image
> 
> Also duplication and complexity notice https://codeclimate.com/github/kodadot/nft-gallery/pull/5345 eyes

1. 23 files are new, some files are changed because i moved things to shared folder, or did simple renaming.

2.the duplication is honestly nothing:

image

3. timeAgo is a simple function. not sure why it is complaining
image

4. getOwners and getFlippers are a bit long that is true, but i don't think they hard to understand

image

**Comment by daiagi** [comment]
> I see some dance at 1090px (two-thirds of 13")

@yangwao  fixed

**Comment by daiagi** [comment]
so... tbh i'm not clear what data should be displayed in the chart.
@yangwao @exezbcz

File: components/collection/utils/timeAgo.ts (hunk 3 above)
Comment (daiagi) [reply components/collection/utils/timeAgo.ts:28]: @Jarsen136  may i have your approval on the PR?

**Comment by yangwao** [comment]
> i'm not clear what data should be displayed in the chart.

current data are ok I guess, we will be tuning chart anyway I guess
- #5362

**Comment by daiagi** [comment]
to sum it up, left overs for next PR:

- [ ] display sensible data in the chart #5362 
- [ ] holder/flipper info on RMRK2
- [ ] use ```formatDistanceFromNow``` instead of ```timeAgo``` https://github.com/kodadot/nft-gallery/pull/5345#discussion_r1148504226

**Review by Jarsen136 — APPROVED** [review]
Some bugfix or code refactor could be moved to the follow-up issue, as this PR is too huge.

**Review by roiLeo — APPROVED** [review]
> Some bugfix or code refactor could be moved to the follow-up issue, as this PR is too huge.

^ this

- I would remove `import { set } from 'vue'`
- make `has-text-k-blue` & `has-text-k-grey` global [variables](https://bulma.io/documentation/customize/variables/) class
- query doesn't have limit on events
- put composable (`useReplaceUrl`) in composable folder 
- relative path
- duplicate css code & some bad selector (`.background-color`??)

**Comment by yangwao** [comment]
pay 150 usd

keeping promise, yet code quality should be kept up! 
- https://github.com/kodadot/nft-gallery/issues/5388#issuecomment-1484910105

I'm leaving this let me know! 
- https://github.com/kodadot/nft-gallery/discussions/5391

**Comment by yangwao** [comment]
let's merge it @vikiival if there are no objections
- work will continue in #5388

**Comment by daiagi** [comment]
> pay 150 usd
> 
> keeping promise, yet code quality should be kept up!
> 
@yangwao 

so no 100 - 200 usd extra?

**Comment by yangwao** [comment]
pay 200 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $200 @ 33.33 USD/KSM ~ 6.001 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x344426e37d86a725362030a807424bbfad06dcf1c96525fdd52f48e054a14c7c](https://kusama.subscan.io/extrinsic/0x344426e37d86a725362030a807424bbfad06dcf1c96525fdd52f48e054a14c7c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by daiagi** [comment]
> pay 200 usd
> 
> seems bot is resting thinking

@yangwao  tbh i expected more, issue is $$ and you said what you said

>yet code quality should be kept up!

code quality is up. the outstanding issues are minor

**Comment by yangwao** [comment]
> issue is $$ and you said what you said

yes, $$ was marked as inclusive within that range. I can do $100 if you would keep you happy ? You will be still up $250 from origin.

**Comment by yangwao** [comment]
Ah I see where confusion happend
- https://github.com/kodadot/nft-gallery/issues/5272#issuecomment-1479504428 -> $100-$200
- $$ means some range 50-150usd which I originally mean as inclusive upper bound for speedy delivery.
- I've sent $200
- do you want $100 to keep you happy for your work?

**Comment by daiagi** [comment]
Sorry @yangwao  I didn't  quite i understand
>  I can do $100 if you would keep you happy

are you offering another 100$? if that's the case then sure and thank you

>You will be still up $250 from origin.

Don't understand what you mean here

**Comment by yangwao** [comment]
pay 100 usd

> are you offering another 100$? if that's the case then sure and thank you
sending another $100! :)

**Comment by daiagi** [comment]
> Ah I see where confusion happend

 yes indeed there it happened.

anyway the matter is done . Thank you @yangwao

**Comment by yangwao** [comment]
> bot stuck again?

yup weird, suspect coingecko api or github webhooks sometimes doesn't deliver, ok seems autocode side

image

**Comment by yangwao** [comment]
pay 100 usd

**Comment by daiagi** [comment]
https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L&usdamount=100

in case you want to it manually this time

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 33.41 USD/KSM ~ 2.993 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x89cf51390a96caf22de24b168162655f6f8d0d425e5fd1257243df00c41c2d64](https://kusama.subscan.io/extrinsic/0x89cf51390a96caf22de24b168162655f6f8d0d425e5fd1257243df00c41c2d64)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by yangwao** [comment]
okay I smell some sort of censorship, might sound very coincidence, but when I've added pay 100 USD into comment, it failed webhook, when I removed it from comment, magically delivered and all @daiagi was delivered in this thread lol sooo, something something is telling me 

image

File: components/collection/activity/Activity.vue [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        </div>
+      </div>
+      <hr class="mb-40" :class="{ 'my-40': !isBreadCrumbsShowing }" />
+      <Events :events="sortedEventsWithOffersDesc" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import ActivityChart from './Chart.vue'
```
Comment (vikiival) [inline components/collection/activity/Activity.vue:31]: Please use relative imports (starting with @)

File: components/collection/activity/Chart.vue [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  }
+)
+
+const toDataPoint = (e: ActivityInteraction): DataPoint => ({
+  timestamp: e.timestamp,
+  value: parseInt(e.meta),
+})
+
+const sortAsc = (events: DataPoint[]) =>
+  events.sort((a, b) => a.timestamp - b.timestamp)
```
Comment (vikiival) [inline components/collection/activity/Chart.vue:29]: Please move functions into utils

File: components/collection/activity/holdersFlippersTabs/moreNFTS.vue [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<template>
```
Comment (vikiival) [inline components/collection/activity/holdersFlippersTabs/moreNFTS.vue:1]: Please use more suitable name for the component 🤗. 
Other components seems to be named correctly

**Comment by vikiival** [comment]
Nice job please fix this and we can proceed

image

File: components/collection/activity/Activity.vue (hunk 5 above)
Comment (daiagi) [reply components/collection/activity/Activity.vue:31]: but Actvity.vue and Chart (renames to ActivityChart to be more clear) are in the same dir.
isn't this acceptable in that case?

File: components/collection/utils/useCollectionActivity.ts [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+              nft,
+            })
+      return
+    }
+
+    // nft isn't consumed and it hasn't change hands => it is owned by it's creator
+    const mintInteraction = events[0]
+    const mintTimeStamp = new Date(mintInteraction.timestamp).getTime()
+
+    owners[nft.currentOwner] =
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/collection/utils/useCollectionActivity.ts:116]: @vikiival  seriously?

File: components/collection/utils/useCollectionActivity.ts [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+        price: parseInt(offer.price),
+        timestamp: new Date(offer.updatedAt).getTime(),
+        interaction: OfferInteraction,
+        nft: { ...nft, events: undefined, offers: undefined },
+      }))
+    )
+    .flat()
+}
+
+const getOwners = (nfts) => {
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/collection/utils/useCollectionActivity.ts:80]: @vikiival I don't think that there will be anything gained by trying to reduce it farther, same goes for getFlippers

File: components/collection/activity/holdersFlippersTabs/moreNFTS.vue (hunk 7 above)
Comment (daiagi) [reply components/collection/activity/holdersFlippersTabs/moreNFTS.vue]: renamed to NFTSDetailsDropdown

File: components/collection/activity/Activity.vue [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import ActivityChart from './ActivityChart.vue'
+import OwnerInsights from './OwnerInsights.vue'
+import Events from './events/Events.vue'
+import BreadcrumbsFilter from '@/components/shared/BreadcrumbsFilter.vue'
+import { useCollectionActivity } from '@/components/collection/utils/useCollectionActivity/useCollectionActivity'
```
Comment (roiLeo) [inline components/collection/activity/Activity.vue:35]:
```suggestion
import { useCollectionActivity } from '@/components/collection/utils/useCollectionActivity/useCollectionActivity'
```
`useCollectionActivity/useCollectionActivity`? IMO all composable should be in the dedicated directory

File: components/collection/activity/events/Events.vue [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+<script setup lang="ts">
+import {
+  InteractionWithNFT,
+  Offer,
+  OfferInteraction,
+} from '@/components/collection/utils/types'
+import EventRow from './EventRow.vue'
+import { useIntersectionObserver } from '@vueuse/core'
+import { Interaction } from '@kodadot1/minimark'
+import { is } from '@/composables/useReplaceUrl'
```
Comment (roiLeo) [inline components/collection/activity/events/Events.vue:43]:
```suggestion
import { is } from '@/composables/useReplaceUrl'
```

`useReplaceUrl` is [auto imported](https://nuxt.com/docs/guide/directory-structure/composables#composables-directory) here

File: components/collection/activity/events/eventRow/EventRowDesktop.vue [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+<style scoped lang="scss">
+.fixed-width {
+  width: 66px;
+}
+.fixed-height {
+  height: 22px;
+}
+.height-50px {
+  height: 50px;
+}
```
Comment (roiLeo) [inline components/collection/activity/events/eventRow/EventRowDesktop.vue:134]: do we need to set fixed height & width?

File: components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Flipper.vue [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+    ready.value = true
+  }
+}
+</script>
+
+<style scoped lang="scss">
+.mr-2rem {
+  margin-right: 1.5rem;
+}
```
Comment (roiLeo) [inline components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Flipper.vue:90]: can't you use Bulma [spacing helpers](https://bulma.io/documentation/helpers/spacing-helpers/)

File: components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Flipper.vue [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+
+.-mx-5 {
+  margin: 0 -1.5rem !important;
+}
```
Comment (roiLeo) [inline components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Flipper.vue:94]: bad negative margin

File: components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Holder.vue [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+</script>
+
+<style scoped lang="scss">
+.mr-2rem {
+  margin-right: 2rem;
+}
+
+.-mx-5 {
+  margin: 0 -1.5rem !important;
+}
```
Comment (roiLeo) [inline components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Holder.vue:73]: this is duplicated

File: components/collection/activity/ownerInsightsTabs/format.ts [hunk 16]
```diff
@@ -0,0 +1,4 @@
+export default function (number: number) {
+  const isInteger = Number.isInteger(number)
+  return isInteger ? number.toString() : number.toFixed(4)
+}
```
Comment (roiLeo) [inline components/collection/activity/ownerInsightsTabs/format.ts:4]: do we need to create a new file for this function?

File: components/shared/filters/modules/PriceFilter.vue [hunk 17]
```diff
@@ -67,7 +67,7 @@
 <script lang="ts" setup>
 import { NeoButton } from '@kodadot1/brick'
 import { fromDecimals, toDecimals } from '@/utils/math'
-import useReplaceUrl from './useReplaceUrl'
+import useReplaceUrl from '@/composables/useReplaceUrl'
```
Comment (roiLeo) [inline components/shared/filters/modules/PriceFilter.vue:70]: should be auto imported

File: components/shared/filters/modules/StatusFilter.vue [hunk 18]
```diff
@@ -28,7 +28,7 @@
 </template>
 
 <script lang="ts" setup>
-import useReplaceUrl from './useReplaceUrl'
+import useReplaceUrl from '@/composables/useReplaceUrl'
```
Comment (roiLeo) [inline components/shared/filters/modules/StatusFilter.vue:31]: auto

File: styles/global.scss [hunk 19]
```diff
@@ -294,3 +297,72 @@ hr {
...
   }
 }
+.border {
+  @include ktheme() {
+    border: 1px solid theme('border-color')
+  }
+}
+
+
+//colors
```
Comment (roiLeo) [inline styles/global.scss:307]: have you seen in `styles/abstracts/variables.scss` it should do the work you looking for

File: components/collection/activity/events/eventRow/EventRowDesktop.vue (hunk 12 above)
Comment (daiagi) [reply components/collection/activity/events/eventRow/EventRowDesktop.vue:134]: long discussion about this with @exezbcz  in discord...
yes

File: styles/global.scss (hunk 19 above)
Comment (daiagi) [reply styles/global.scss:307]: honestly i could not figure it out

File: components/collection/activity/ownerInsightsTabs/format.ts (hunk 16 above)
Comment (daiagi) [reply components/collection/activity/ownerInsightsTabs/format.ts]: move it to utils

File: components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Flipper.vue (hunk 13 above)
Comment (daiagi) [reply components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Flipper.vue]: yes. done

File: components/collection/activity/Activity.vue (hunk 10 above)
Comment (daiagi) [reply components/collection/activity/Activity.vue]: done

File: components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Flipper.vue (hunk 14 above)
Comment (daiagi) [reply components/collection/activity/ownerInsightsTabs/NFTSDetailsDropdown/Flipper.vue]: removed

File: components/shared/filters/modules/EventTypeFilter.vue [hunk 20]
```diff
@@ -0,0 +1,122 @@
+<template>
+  <b-collapse
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/shared/filters/modules/EventTypeFilter.vue:2]: moving it to the follow up issue with your permission

File: components/collection/activity/events/eventRow/EventRowDesktop.vue (hunk 12 above)
Comment (daiagi) [reply components/collection/activity/events/eventRow/EventRowDesktop.vue:137]: altough fixed height can probably be dropped

Comment (exezbcz) [reply components/collection/activity/events/eventRow/EventRowDesktop.vue:137]: is that for the tags? :D

Comment (daiagi) [reply components/collection/activity/events/eventRow/EventRowDesktop.vue:137]: dropping fixed height causes height diff between desktop and tablet view
i guess min-height could solve it, but what would be the point of that then?

**Comment by yangwao** [comment]
image

**Comment by yangwao** [comment]
Continue @daiagi here
- https://github.com/kodadot/nft-gallery/issues/5362

**Comment by daiagi** [comment]
> Continue @daiagi here

Isn't it p4?

**Pull request merged**